lib/msf/base/logging.rb

```diff
@@ -23,15 +23,19 @@ def self.init
 
       f = Rex::Logging::Sinks::Flatfile.new(
         Msf::Config.log_directory + File::SEPARATOR + "framework.log")
-
+      error_f = Rex::Logging::Sinks::Flatfile.new(
+        Msf::Config.log_directory + File::SEPARATOR + "error.log")
+      /(\[\d\d\/\d\d\/\d\d\d\d \d\d:\d\d:\d\d\] \[.*?\] error:(?:(?!\n\n).)+)/
       # Register each known log source
       [
         Rex::LogSource,
         Msf::LogSource,
-        'base',
+        'base'
       ].each { |src|
         register_log_source(src, f)
       }
+
+      register_log_source('error', error_f)
```

Comment:
I think this will lead to good questions for module hacking 👀

Reply:
![sweat](https://user-images.githubusercontent.com/54621924/81931224-a9f7f000-95e1-11ea-8a8d-dcd5123766c0.gif)